cli/slsa-verifier/verify/verify_npm_package.go

```diff
@@ -74,22 +75,22 @@ func (c *VerifyNpmPackageCommand) Exec(ctx context.Context, tarballs []string) (
 
 		attestations, err := os.ReadFile(c.AttestationsPath)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Verifying npm package %s: FAILED: %v\n\n", tarball, err)
+			slog.Error(fmt.Sprintf("Verifying npm package %s: FAILED: %v\n\n", tarball, err))
 			return nil, err
 		}
 
 		verifiedProvenance, outBuilderID, err := verifiers.VerifyNpmPackage(ctx, attestations, tarballHash, provenanceOpts, builderOpts)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Verifying npm package %s: FAILED: %v\n\n", tarball, err)
+			slog.Error(fmt.Sprintf("Verifying npm package %s: FAILED: %v\n\n", tarball, err))
 			return nil, err
 		}
 
 		if c.PrintProvenance {
-			fmt.Fprintf(os.Stdout, "%s\n", string(verifiedProvenance))
+			slog.Info(fmt.Sprintf("%s\n", string(verifiedProvenance)))
 		}
 
 		builderID = outBuilderID
-		fmt.Fprintf(os.Stderr, "Verifying npm package %s: PASSED\n\n", tarball)
```

Comment:
where does slog write the result? We need to ensure compatibility and ensure it's written to stderr when `fmt.Fprintf(os.Stderr`. Otherwise folks won't be able to pipe the result into jq or other tools: jq will pick not just the printed provenance, but the other messages too.

I think before changing this API we should define what the interface is for our logger. And we should provide an interface as logger, so that callers can customize how they log. Here's a fast logging library https://github.com/uber-go/zap, among others, that would need to be instantiable by callers

Reply:
I confirmed that by default slog will send to stderr. And then I removed the changes against `fmt.Fprintf(os.Stdout, ...` statements.

I'll look into how to best allow caller-provided loggers.

Reply:
FWIW, I'm personally in favor of using the stdlib logging packages rather than including a dependency.